src/shady-css/parser.js

```diff
@@ -0,0 +1,229 @@
+import { matcher, tokenType } from './common';
+import { TokenCursor } from './token-cursor';
+import { Lexer } from './lexer';
+import { NodeFactory } from './node-factory';
+
+class Parser {
+  constructor(nodeFactory = new NodeFactory()) {
+    this.nodeFactory = nodeFactory;
+  }
+
+  parse(cssText) {
+    return this.parseStylesheet(new TokenCursor(new Lexer(cssText)));
+  }
+
+  parseStylesheet(tokenCursor) {
+    let declarations = [];
+    let declaration;
+
+    while (tokenCursor.next &&
+           (declaration = this.parseDeclaration(tokenCursor))) {
+      declarations.push(declaration);
+    }
+
+    return this.nodeFactory.stylesheet(declarations);
+  }
+
+  parseDeclaration(tokenCursor) {
+    // Consume all leading (presumably insignificant) whitespace:
+    this.traverseWhitespace(tokenCursor);
```

Comment:
I'd consume the 0 or 1 whitespace tokens before this in parseStylesheet. parseDeclaration should assume it's starting on the first token of the declaration.